src/components/Shared/SearchComponent.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  Dispatch,
+  SetStateAction,
+  useEffect,
+  useState,
+  createRef,
+} from 'react';
+
+import CheckIcon from '@mui/icons-material/Check';
+import CloseIcon from '@mui/icons-material/Close';
+import Button from '@mui/material/Button';
+import Grid from '@mui/material/Grid';
+import useMediaQuery from '@mui/material/useMediaQuery';
+
+import { RootState } from '../../common/store';
+import { useAppDispatch } from '../../hooks/app';
+import { useAppSelector } from '../../hooks/app';
+import useSelectRevision from '../../hooks/useSelectRevision';
+import { clearCheckedRevisions } from '../../reducers/CheckedRevisions';
+import { SearchStyles } from '../../styles';
+import type { RevisionsList, InputType } from '../../types/state';
+import SearchDropdown from '../Search/SearchDropdown';
+import SearchInput from '../Search/SearchInput';
+import SearchResultsList from '../Search/SearchResultsList';
```

Comment:
It's in the Shared folder because the Results view will eventually use this Search Component too, so it makes sense to leave it in Shared?

Reply:
I'd say that either you put everything in `/Shared/Search`, or everything in `/Search`.

The rationale is that we try to have imports between directories work in one direction only, to avoid "spaghetti". Shared libraries are usually at the bottom of the import tree, they don't import from other directories (except maybe a `utils` directory that usually contains low-level shared code) exactly because other directories import from `shared`.

Also, I tend to think of a `shared` library as a bag of things that are independent with each other, and don't warrant their own directory in a sense. I know developers that don't want a `shared` directory at all and prefer to group their shared components by bigger topics.

In the case of the `Search` component, I think it definitely warrants its own directory, with many other dependency files. Being in its own directory `/Search` doesn't prevent it from being imported by different other components!

I strongly believe that Search and its dependencies should be together, but in the end I'll let you decide if you want everything in `components/Shared` or in `components/`.

Reply:
Well, yeah, I know putting it in `Search` won't stop it from being imported, of course, lol. However, the rationale makes sense so I don't mind moving it out of 'Shared' and keeping all Search components in one directory and so on.

Reply:
Will follow up in next PR.

Reply:
I see that SearchComponent is loaded in CompareBase, so after all it might make sense to move all Search components to a Shared subdirectory. Looking at this in a separate PR sounds good to me, as this may need more changes than just moving things around.